### PR TITLE
Add analytics for opening the view, executing actions, and navigation.

### PR DESCRIPTION
**Pull request by scheglov**
closes https://github.com/flutter/flutter-intellij/issues/1716

### CODE REVIEW

File: src/io/flutter/preview/PreviewView.java [hunk 1]
```diff
@@ -172,16 +173,18 @@ public void loadState(PreviewViewState state) {
   }
 
   public void initToolWindow(@NotNull ToolWindow toolWindow) {
+    sendAnalyticEvent("initToolWindow");
```
Comment (devoncarew) [inline src/io/flutter/preview/PreviewView.java]: I think we're already getting this event from listening to general view changes; I think we can avoid sending this specific one in order to match what we do for other view (unless we think this adds some specific value).

File: src/io/flutter/preview/PreviewView.java [hunk 2]
```diff
@@ -616,14 +628,22 @@ private void scrollTreeToNodeOnTop(TreeNode node) {
     }
   }
 
+  private void sendAnalyticEvent(@NotNull String name) {
+    FlutterInitializer.getAnalytics().sendEvent("flutterPreview", name);
```
Comment (devoncarew) [inline src/io/flutter/preview/PreviewView.java]: `flutterPreview` ==> `preview`?

File: src/io/flutter/preview/PreviewView.java [hunk 3]
```diff
@@ -616,14 +628,22 @@ private void scrollTreeToNodeOnTop(TreeNode node) {
...
+
+    QuickAssistAction(@NotNull String id, Icon icon, String assistMessage) {
       super(assistMessage, null, icon);
+      this.id = id;
       messageToActionMap.put(assistMessage, this);
     }
 
     @Override
     public void actionPerformed(AnActionEvent e) {
+      sendAnalyticEvent(id);
```
Comment (devoncarew) [inline src/io/flutter/preview/PreviewView.java]: 👍

File: src/io/flutter/preview/PreviewView.java [hunk 4]
```diff
@@ -291,6 +294,15 @@ private void selectPath(TreePath selectionPath, boolean focusEditor) {
       return;
     }
 
+    if (focusEditor) {
```
Comment (devoncarew) [inline src/io/flutter/preview/PreviewView.java]: rename both to 'jumpToSource'? 'showSource'?

File: src/io/flutter/preview/PreviewView.java (hunk 1 above)
Comment (scheglov) [reply src/io/flutter/preview/PreviewView.java]: OK, removing this.

File: src/io/flutter/preview/PreviewView.java (hunk 2 above)
Comment (scheglov) [reply src/io/flutter/preview/PreviewView.java]: I was not sure if we get enough information to disambiguate it from other possible previews.
Fixed.

File: src/io/flutter/preview/PreviewView.java (hunk 4 above)
Comment (scheglov) [reply src/io/flutter/preview/PreviewView.java]: Not sure which "both".
I speculatively changed the name of the parameter and the name of the analytics event.
PTAL

Comment (devoncarew) [reply src/io/flutter/preview/PreviewView.java]: Sorry, I meant both `jumpToSourceDartOutline` and `jumpToSourceFlutterOutline` could just be simplified to `jumpToSource` - that we don't need to differentiate between the source element type.

**Pull request merged**